src/Hl7.Fhir.Core.Tests/Rest/Mocks/MockHttpMessageHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Hl7.Fhir.Core.Tests.Rest.Mocks
+{
+    public class MockHttpMessageHandler : HttpClientHandler
```

Comment:
This MessageHandler could be used by people who want to still use the OnBefore/OnAfter logic, right?  The name "mock" did not suggest that to me (I would think it has something to do with testing).  Maybe call it "RequestResponseEventMessageHandler" or something like that?

Reply:
Oh, this is actually there to support testing ;-)   But we could provide it more generally, as I mentioned, right?

Reply:
Yes, this could be provided more generally. I will move this to the new namespace and rename.